app/stores/data-source.ts

```diff
@@ -0,0 +1,62 @@
+import create, { StateCreator, StoreApi } from "zustand";
+
+import { DataSource } from "@/configurator";
+import {
+  DEFAULT_DATA_SOURCE,
+  getURLParam,
+  parseDataSource,
+  stringifyDataSource,
+} from "@/domain/datasource";
+import { isRunningInBrowser } from "@/lib/is-running-in-browser";
+
+type DataSourceStore = {
+  dataSource: DataSource;
+  setDataSource: (value: string) => void;
+};
+
+const customPersist =
+  (name: string, config: StateCreator<DataSourceStore>) =>
+  (
+    set: StoreApi<DataSourceStore>["setState"],
+    get: StoreApi<DataSourceStore>["getState"],
+    api: StoreApi<DataSourceStore>
+  ) => {
+    const state = config(
+      (payload: DataSourceStore) => {
+        set(payload);
+
+        if (isRunningInBrowser()) {
+          localStorage.setItem(name, stringifyDataSource(payload.dataSource));
+        }
+      },
+      get,
+      api,
+      []
+    );
+
+    let dataSource = DEFAULT_DATA_SOURCE;
+
+    if (isRunningInBrowser()) {
+      const urlDataSource = getURLParam("dataSource");
```

Comment:
Shouldn't it be `name` instead of `"dataSource"` ?

It seems the persist is a bit generic (for localStorage) but not fully.